vmm/src/lib.rs

```diff
@@ -491,17 +497,28 @@ impl EpollContext {
         (dispatch_base, sender)
     }
 
+    // See the below comment for `allocate_virtio_net_tokens`, for an explanation on the returned
+    // values.
     fn allocate_virtio_block_tokens(&mut self) -> (virtio::block::EpollConfig, usize) {
         let (dispatch_base, sender) = self.allocate_tokens(virtio::block::BLOCK_EVENTS_COUNT);
         (
             virtio::block::EpollConfig::new(dispatch_base, self.epoll_raw_fd, sender),
-            self.device_handlers.len(),
+            self.device_handlers.len() - 1,
         )
     }
 
-    fn allocate_virtio_net_tokens(&mut self) -> virtio::net::EpollConfig {
+    // Horrible, horrible hack, because velocity: return a tuple (epoll_config, handler_idx),
+    // since, for some reason, the VMM doesn't own and cannot contact its live devices. I.e.
```

Comment:
Well it's not for some reason, it's for the exact reason you explained :smile: Ownership is transferred, what can you do.

Reply:
Eh.. what can you do...?